lib/vagrant/util/downloader.rb

```diff
@@ -203,9 +203,13 @@ def execute_curl(options, subprocess_options, &data_proc)
           @logger.warn("Downloader exit code: #{result.exit_code}")
           parts    = result.stderr.split(/\n*curl:\s+\(\d+\)\s*/, 2)
           parts[1] ||= ""
-          raise Errors::DownloaderError,
-            code: result.exit_code,
-            message: parts[1].chomp
+          if parts[1].include? "416"
+            # All good actually. 416 means there is no mory bytes to download
```

Comment:
Moreover I actually tested changes in the code as per my issue #7885. In that case file was completely and correctly downloaded but box was not added, because 416 returned which is treated currently as download failure. With this patch it unpacks the file and box is successfully added.